src/containers/unicode_point_map.rs

```diff
@@ -418,13 +422,28 @@ impl UnicodePointMap {
 }
 
 /// Iterator for [`UnicodePointMap`].
+#[derive(Debug)]
 pub struct UnicodePointMapIter<'a> {
     /// Reference to the providing [`UnicodePointMap`] instance.
     upm : &'a UnicodePointMap,
     /// (Optional) index into the vector.
     vec_index : Option<usize>,
     /// (Optional) interator into the map.
     map_iter : Option<std::collections::hash_map::Iter<'a, char, isize>>,
+    /// iteration index (required to support `ExactSizeIterator`)
+    iix : usize,
+}
+
+impl ExactSizeIterator for UnicodePointMapIter<'_> {
+    /// Obtains the remaining number of records accessible via the iterator
+    /// instance.
+    #[inline]
+    fn len(&self) -> usize {
```

Comment:
A default implementation is provided assuming that `Iterator::size_hint` is implemented correctly to give the exact bounds.

I think the better way is to implement `Iterator::size_hint` with the exact bounds and then just rely on the default implementation of `ExactSizeIterator::len`.

You want to implement `Iterator::size_hint` anyway since some allocation optimizations depend on it.